tests/integration_python/test_exec.py

```diff
@@ -0,0 +1,24 @@
+from pathlib import Path
+from .common import verify_cli_command
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+
+def test_run_in_shell_environment(pixi: Path, dummy_channel_1: str) -> None:
+    with ThreadPoolExecutor(max_workers=2) as executor:
```

Comment:
I think we need to use `ProcessPoolExecutor` as we would like them to run in parallel and not in concurrent

https://docs.python.org/3/library/concurrent.futures.html#processpoolexecutor